Prepare some Geometry ESProducers for concurrent IOVs

wddgit opened the pull request:
The following types of changes are included:
    - changing return type of produce to unique_ptr
    - removing unnecessary "dependsOn" callbacks
    - some callback content moved to produce
    - 4 callbacks were simply empty (did nothing)
    - some objects do not need to be cached create on stack
    - removing unneeded data members, includes, declarations
    - fix potential assert failure if produce called more than once
    - fix potential double delete if produce called more than once
    - 2 memory leaks
    - some general cleanup deleting unnecessary things

wddgit committed:
Prepare some Geometry ESProducers for concurrent IOVs

The following types of changes are included:
    - changing return type of produce to unique_ptr
    - removing unnecessary "dependsOn" callbacks
    - some callback content moved to produce
    - 4 callbacks were simply empty (did nothing)
    - some objects do not need to be cached create on stack
    - removing unneeded data members, includes, declarations
    - fix potential assert failure if produce called more than once
    - fix potential double delete if produce called more than once
    - 2 memory leaks
    - some general cleanup deleting unnecessary things

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24612/6505

cmsbuild commented:
A new Pull Request was created by  @wddgit (W. David Dagenhart) for master.

It involves the following packages:

Geometry/EcalMapping
Geometry/HcalCommonData
Geometry/HcalEventSetup
Geometry/HcalTowerAlgo
Geometry/MuonNumbering


@civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please review it and eventually sign? Thanks.
@ptcox, @argiro this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

wddgit commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9ab1157b373f84768a59f2b8b4793ccf0e544b4c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24612/30538/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24612/30538/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3146746
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3146548
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

ianna commented on Geometry/HcalCommonData/plugins/HcalDDDSimConstantsESModule.cc:
```diff
@@ -23,36 +23,25 @@
   ~HcalDDDSimConstantsESModule() override;
 
-  typedef std::shared_ptr<HcalDDDSimConstants> ReturnType;
+  typedef std::unique_ptr<HcalDDDSimConstants> ReturnType;
```
```
using ReturnType = std::unique_ptr<HcalDDDSimConstants>;
```

ianna commented on Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h:
```diff
@@ -9,22 +9,18 @@
-// class EcalMappingRcd;
-
-
 // #include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
```
please, remove commented out includes

ianna commented on Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h:
```diff
@@ -9,22 +9,18 @@
-
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
 // #include "Geometry/Records/interface/IdealGeometryRecord.h"
```
please, remove commented out includes

ianna commented on Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h:
```diff
@@ -41,14 +37,10 @@ class EcalElectronicsMappingBuilder : public edm::ESProducer
   
-  
-  const EcalMappingElectronics* Mapping_ ;
   // void setIntervalFor(const edm::eventsetup::EventSetupRecordKey &, const edm::IOVSyncValue&, edm::ValidityInterval & );
```
please, remove commented out code

ianna commented on Geometry/HcalEventSetup/interface/HcalDDDGeometryEP.h:
```diff
@@ -27,18 +27,15 @@ class HcalDDDGeometryEP : public edm::ESProducer {
   ~HcalDDDGeometryEP() override;
 
-  typedef std::shared_ptr<CaloSubdetectorGeometry> ReturnType;
+  typedef std::unique_ptr<CaloSubdetectorGeometry> ReturnType;
```
```
using ReturnType = std::unique_ptr<CaloSubdetectorGeometry>;
```

ianna commented on Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h:
```diff
@@ -19,13 +19,11 @@ class HcalHardcodeGeometryEP : public edm::ESProducer {
   ~HcalHardcodeGeometryEP() override;
 
-  typedef std::shared_ptr<CaloSubdetectorGeometry> ReturnType;
+  typedef std::unique_ptr<CaloSubdetectorGeometry> ReturnType;
```
```
using ReturnType = std::unique_ptr<CaloSubdetectorGeometry>;
```

ianna commented on Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h:
```diff
@@ -19,13 +19,11 @@ class HcalHardcodeGeometryEP : public edm::ESProducer {
-  void       idealRecordCallBack(const HcalRecNumberingRecord&) {}
-
 private:
   edm::ParameterSet ps0;
```
```
edm::ParameterSet ps0_;
```

ianna commented on Geometry/HcalEventSetup/interface/HcalTopologyIdealEP.h:
```diff
@@ -28,18 +28,15 @@ class HcalTopologyIdealEP : public edm::ESProducer {
   ~HcalTopologyIdealEP() override;
 
-  typedef std::shared_ptr<HcalTopology> ReturnType;
+  typedef std::unique_ptr<HcalTopology> ReturnType;
```
```
using ReturnType = std::unique_ptr<HcalTopology>;
```

ianna commented on Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc:
```diff
@@ -39,19 +35,11 @@ class MuonNumberingInitialization : public edm::ESProducer
   typedef std::unique_ptr<MuonDDDConstants> ReturnType;
```
```
using ReturnType = std::unique_ptr<MuonDDDConstants>;
```

wddgit committed:
Additional cleanup including:

  - deleting commented code and useless comments
  - deleting trailing whitespace
  - converting from typedef to using statements
  - removing an unused copy of a ParameterSet
  - removing unneeded blank lines

Initially motivated by Ianna's comments. I fixed all
those and did more. None of this should change any behavior

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24612/6525

cmsbuild commented:
Pull request #24612 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

wddgit commented:
please test
I pushed a new fix that cleans up all the problems Ianna suggested I fix. Thanks for the suggestions. When I got going I also fixed some other similar things. So the second commit includes:

      - deleting commented code and useless comments
      - deleting trailing whitespace
      - converting from typedef to using statements
      - removing an unused copy of a ParameterSet
      - removing unneeded blank lines

Unless I made a typographical blunder, none of this should effect behavior in any way, although it does make the code easier to read.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b3ecea400cca533888001c0ee3a8dbefe9aa5def
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24612/30563/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24612/30563/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3146746
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3146548
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

ianna commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
+1
